api/backend/blueflood/blueflood.go

```diff
@@ -198,8 +198,56 @@ func bluefloodResolution(r int64) string {
 	return Resolution1440Min
 }
 
-func resample(points []float64, currentResolution int64, expectedTimerange api.Timerange, sampleMethod api.SampleMethod) ([]float64, error) {
-	return nil, errors.New("Not implemented")
+// Resamples the original data returned from Blueflood. Returns raw values
+// corresponding to the `timerange`.
+//
+// Snaps data left to the `timerange` buckets. Resampling buckets using data
+// to the right of the timestamp (i.e. the data point at time 123000 with
+// resolution 60 is generated from [123000, 123060). If there is no data, that
+// point is filled with NaN (note: this means upsampling doesn't work).
+//
+// Assumes `points` is sorted chronologically.
+func resample(points []MetricPoint, timerange api.Timerange, sampleMethod api.SampleMethod) ([]float64, error) {
+	series := make([]float64, timerange.Slots())
+
+	// Cursor into `series`
+	idx := 0
+	// Cursors into `points` for working data windows
+	windowStart, windowEnd := 0, 0
+	// Generate points at timestamps corresponding to the timerange
+	for t := timerange.Start(); t <= timerange.End(); t, idx = t+timerange.Resolution(), idx+1 {
+		// Set up our series window
+		for ; points[windowStart].Timestamp < t; windowStart++ {
```

Comment:
i think here the assumption is `points[...].Timestamp` is between `[Start(), End()]`

it's a bit subtle, maybe add a condition for `windowStart < Slots`?